### ISSUE TITLE
[Warning] TStreamerInfo.cxx: ‘this’ pointer is null [-Wnonnull]

### ISSUE THREAD

**Issue by kgizdov** [root-project/root#8981]
- [x] Checked for duplicates

### Describe the bug
Warning during build time ROOT 6.24.06, GCC 11.1, Arch Linux x86_64
```
[ 77%] Built target cppyy3_9
/build/root/src/root-6.24.06/io/io/src/TStreamerInfo.cxx: In member function ‘virtual void TStreamerInfo::BuildCheck(TFile*, Bool_t)’:
/build/root/src/root-6.24.06/io/io/src/TStreamerInfo.cxx:1110:29: warning: ‘this’ pointer is null [-Wnonnull]
 1110 |                      Warning("BuildCheck", "\n\
      |                      ~~~~~~~^~~~~~~~~~~~~~~~~~~
 1111 |    The StreamerInfo from %s does not match existing one (%s:%d)\n\
      |    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1112 |    The existing one has not been used yet and will be discarded.\n\
      |    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1113 |    Reading should work properly, however writing object of\n\
      |    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1114 |    type %s will not work properly.  Most likely the version number\n\
      |    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1115 |    of the class was not properly updated [See ClassDef(%s,%d)].",
      |    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1116 |                              file->GetName(), GetName(), fClassVersion, GetName(), GetName(), fClassVersion);
      |                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[ 77%] Linking CXX shared library ../../lib/libRIO.so
```
### Expected behavior
No null pointers.

### To Reproduce
1. Build ROOT
2. See warning

### Setup
1. ROOT version: 6.24.06
2. Operating system: Arch Linux x86_64
3. How you obtained ROOT: GitHub, built from source

### Additional context
Build options located [here](https://github.com/archlinux/svntogit-community/tree/packages/root/trunk)

**Comment by pcanal** [comment]
I am confused ... how can the `this` pointer be null in a virtual function?  Why does it complains only about this line (and not all the other lines above and below)?  Is there some context missing (and the problem is really a caller of the `TStreamerInfo::Build` function)?

**Comment by kgizdov** [comment]
so I see a similar warning in many other places this function is called/overloaded, etc. Not sure about the pattern, but it might be that the compiler can unfold the full vtable and the base object is always null or something of the sort. I will have a deeper dive to check.

**Comment by kgizdov** [comment]
oh, wait, this is fixed in #8351

**Comment by kgizdov** [comment]
oh, wait, this is fixed in #8351

**Comment by jalopezg-r00t** [comment]
Duplicate of #8129.  Thanks for reporting, though!
